internal/httpclient/client_socks5_test.go

```
@@ -0,0 +1,217 @@
+
+	// Stand in for net/http.Transport dialing the CONNECT tunnel's proxy
+	// address directly (never "example.com:443") with a real, reachable
+	// listener. dialWithDecision must reach it directly -- if the dial
+	// decision were (wrongly) re-derived from this address instead of
+	// read from ctx, HTTP_PROXY's bogus SOCKS5 proxy would apply instead
+	// and this dial would fail or hang rather than connecting straight in.
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ln.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `ln.Close` is not checked

(errcheck)